app/src/Controller/ArticlesController.php

```diff
@@ -78,4 +93,40 @@ public function delete($slug)
             return $this->redirect(['action' => 'index']);
         }
     }
+
+    public function tags()
+    {
+        // The 'pass' key is provided by CakePHP and contains all
+        // the passed URL path segments in the request.
+        $tags = $this->request->getParam('pass');
+
+        // Use the ArticlesTable to find tagged articles.
+        $articles = $this->Articles->find('tagged', [
+                'tags' => $tags
+            ])
+            ->all();
+
+        // Pass variables into the view template context.
+        $this->set([
+            'articles' => $articles,
+            'tags' => $tags
+        ]);
+    }
+
+    /* getting from params
+    public function tags(...$tags)
+    {
+        // Use the ArticlesTable to find tagged articles.
+        $articles = $this->Articles->find('tagged', [
+                'tags' => $tags
+            ])
+            ->all();
+
+        // Pass variables into the view template context.
+        $this->set([
+            'articles' => $articles,
+            'tags' => $tags
+        ]);
+    }
+    */
```

Comment:
Is it necessary to leave the unwanted code here?

Reply:
no, it was removed